File: abc/abc164/abc164_c.fs
// https://atcoder.jp/contests/abc164/submissions/12399892

let N = stdin.ReadLine() |> int

let S = Array.zeroCreate<string> N

for i in [ 0..N - 1 ] do
    S.[i] <- stdin.ReadLine()

S
|> Array.distinct
|> Array.length
|> stdout.WriteLine

File: abc/abc164/abc164_c.cs
// https://atcoder.jp/contests/abc164/submissions/12399846

using System;
using System.Linq;

class Program
{
    static void Main()
    {
        var N = int.Parse(Console.ReadLine());

        var S = new string[N];
        for (int i = 0; i < N; i++)
        {
            S[i] = Console.ReadLine();
        }

        Console.WriteLine(S.Distinct().Count());
    }
}
